### ISSUE TITLE
Ranges shows full calendar when date is specified as text

### ISSUE THREAD

**Issue by subkams** [dangrossman/bootstrap-daterangepicker#1397]
Hi,
Must congratulate you on a very nice product. 
Lately, I had a requirement to show Custom Ranges and I tried the code using Dates as text.
 But this would bring up the full calendar instead of Only the Range Names.
If only [moment() ,moment()] is used it works as expected.

I am using the following code
options.ranges = {
        'now':[moment(),moment()],
	'Today': [moment("09-08-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
        'Tomorrow': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
        'dayafter': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
        'sat': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
        'sun': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
      
Rest of the settings are as per demo. 
Can you suggest the correct formatting as it doesnt seem to be documented anywhere.

![screencapture-file-c-users-subha_000-downloads-bootstrap-daterangepicker-master-bootstrap-daterangepicker-master-example-amd-index-html-1487436836044](https://cloud.githubusercontent.com/assets/108209/23095039/41802d68-f629-11e6-9216-8ee4c41b7916.png)

**Comment by dangrossman** [comment]
This is working as intended. I do not see the issue. The calendars should be visible if a custom range, not a predefined range, is selected.

**Comment by subkams** [comment]
Hi Dan, 
I am sorry , I used the wrong terminology. 
I actually meant Specific Ranges (not a Custom Range) . 

I used your config builder and modified the code to add the ranges as below. 
options.ranges = {
'now':[moment(),moment()],
'Today': [moment("09-08-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
'Tomorrow': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
'dayafter': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
'sat': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],
'sun': [moment("09-05-2017","MM-DD-YYYY"), moment("09-05-2017","MM-DD-YYYY")],

I want to display only these Range Names: Now, Today,Tomorrow...
I dont want to display the calendar. 
It seems to work when only moment() is used. 
The above variation of moment() ( as given in moment.js site ) results in the calendar being shown. 
That too it happens only the first time you click on the input box.
Next time onwards it shows the popup as expected i.e with No Calendar.
So is there a setting or a correct way to pass dates in string format.

Thanks & Regards
